src/util/castJsonTypes.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Values in the JSON should be appropriately typed however this may not always be the case.
+ * For instance, certain values of a continuous trait may be strings, and we should cast
+ * these to numbers. See https://github.com/nextstrain/auspice/issues/1626 for more discussion
+ * of this particular case. Feel free to add more checks / casts to this function!
+ * @param {Object} metadata
+ * @param {Object} tree
+ * @returns {undefined} Any type casting is in-place
+ */
+export const castIncorrectTypes = (metadata, tree) => {
+  try {
+    const continuousKeys = new Set();
+    Object.entries(metadata.colorings || {}).forEach(([key, details]) => {
+      if (details.type==="continuous") {
+        continuousKeys.add(key);
+      }
+    });
+    tree.nodes.forEach((node) => {
+      for (const key of continuousKeys) {
+        const attr = node?.node_attrs?.[key];
+        if (!attr) continue;
+        toNumber(attr);
+      }
+    });
+  } catch (err) {
+    // type casting shouldn't be required (the JSON should be correctly typed)
+    // so any errors shouldn't prevent Auspice loading
+    console.error(err);
+  }
+};
+
+export function toNumber(attr) {
```

Comment:
_nitpick_

I would rename this function to be more specific, e.g. `continuousAttrValueToNumber`, since it designed to only be used in this specific case.

Reply:
Done in [eca540e](https://github.com/nextstrain/auspice/pull/1655/commits/eca540e209ffcc56121b260c9ec5152b08e7a0a2)